platforms/core-execution/file-watching/src/main/java/org/gradle/internal/watch/registry/impl/AbstractFileWatcherUpdater.java

```diff
@@ -56,9 +56,10 @@ public AbstractFileWatcherUpdater(
     }
 
     @Override
-    public void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarchy root) {
+    public void registerWatchableHierarchy(File watchableHierarchy, SnapshotHierarchy root, File probeLocation) {
         watchableHierarchies.registerWatchableHierarchy(watchableHierarchy, root);
-        probeRegistry.registerProbe(watchableHierarchy);
+        // TODO: probably it would be better to have a service that maps hierarchy to probe location with respect to the project-cache-dir instead of passing it through
```

Comment:
🤔 Isn't that what you are turning the probe registry into just now? What do you mean "with respect to the project-cache-dir"?

Or do you mean instead of raw `File` hierarchies we should be passing around some higher level thing that would allow us to look up the project cache dir from?